src/software/thunderscope/robot_communication.py

```diff
@@ -81,27 +94,37 @@ def run(self):
         that the robots timeout and stop.
 
         NOTE: If disconnect_fullsystem_from_robots is called, then the packets
-        will not be forwarded to the robots. 
+        will not be forwarded to the robots.
 
         send_override_primitive_set can be used to send a primitive set, which
         is useful to dip in and out of robot diagnostics.
 
         """
         while True:
-            if self.fullsystem_connected_to_robots:
-
-                # Send the world
-                world = self.world_buffer.get(block=True)
-                self.world_mcast_sender.send_proto(world)
+            # total primitives for all robots
+            robot_primitives = dict()
 
-                # Send the primitive set
-                primitive_set = self.primitive_buffer.get(block=False)
+            # fullsystem is running, so world data is being received
+            if self.robots_connected_to_fullsystem:
+                diagnostics_primitive = DirectControlPrimitive(
+                    motor_control=self.motor_control_diagnostics_buffer.get(
+                        block=False
+                    ),
+                    power_control=self.power_control_diagnostics_buffer.get(
+                        block=False
+                    ),
+                )
```

Comment:
if we're connected to full system, we shouldn't be getting primitives from diagnostics

Reply:
sorry I added this for testing but forgot to remove